### PR TITLE
Add yum lock/unlock around update check

### PR DESCRIPTION
**Pull request by utopiabound**
Ensure yum integrity via yum lock.

Signed-off-by: Nathaniel Clark <nclark@whamcloud.com>

### CODE REVIEW

File: iml-update-check.py [hunk 1]
```diff
@@ -41,6 +42,8 @@
         has_updates |= len(ypl.updates) > 0
 
 yp.close()
+yp.closeRpmDB()
+yp.doUnlock()
```
Comment (jgrund) [inline iml-update-check.py]: What happens if the code before the unlock has an error, do we stay locked?

Comment (utopiabound) [reply iml-update-check.py]: Could wrap everything in try/except

Comment (utopiabound) [reply iml-update-check.py]: actually YumBase on __del__ will run these three commands, so we don't actually NEED to call them.

Comment (jgrund) [reply iml-update-check.py]: Is this patch effectively a noop then?

Comment (utopiabound) [reply iml-update-check.py]: No, the lock at the top is a good thing. And `__del__()` is gaurenteed to be called when the interpreter exits, if there's still a reference to an object.

Comment (jgrund) [reply iml-update-check.py]: So, the yum api doesn't take a lock when a new yum object is created?

Comment (utopiabound) [reply iml-update-check.py]: Not that I see.

**Pull request merged**